### PR TITLE
[MI-1893]Disable prejoin page

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
1. Added radio button in settings to disable/enable prejoin page.
2. Fixed linting errors in go structs.

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -34,6 +34,12 @@
                 "type": "bool",
                 "help_text": "(Experimental) When true, Jitsi video is embedded as a floating window inside Mattermost by default. Users can override this setting with '/jitsi settings'."
             },
+            {
+                "key": "JitsiPrejoinPage",
+                "display_name": "Show prejoin page:",
+                "type": "bool",
+                "help_text": "When true, prejoin page will not be displayed when meet is embedded inside mattermost."
```
Comment (manojmalik20) [inline plugin.json]:
```suggestion
                "help_text": "When false, pre-join page will not be displayed when Jitsi is embedded inside Mattermost."
```

File: plugin.json [hunk 2]
```diff
@@ -34,6 +34,12 @@
                 "type": "bool",
                 "help_text": "(Experimental) When true, Jitsi video is embedded as a floating window inside Mattermost by default. Users can override this setting with '/jitsi settings'."
             },
+            {
+                "key": "JitsiPrejoinPage",
+                "display_name": "Show prejoin page:",
```
Comment (manojmalik20) [inline plugin.json]:
```suggestion
                "display_name": "Show pre-join page:",
```

File: webapp/src/components/conference/conference.tsx [hunk 3]
```diff
@@ -101,6 +103,10 @@ export default class Conference extends React.PureComponent<Props, State> {
             onload: () => {
                 this.setState({loading: false});
                 this.resizeIframe();
+            },
+            configOverwrite: {
+                // disable the prejoin page
```
Comment (manojmalik20) [inline webapp/src/components/conference/conference.tsx]:
```suggestion
                // Disable the pre-join page
```

File: webapp/src/components/conference/index.ts [hunk 4]
```diff
@@ -8,10 +8,14 @@ import Conference from './conference';
 import {openJitsiMeeting, setUserStatus} from '../../actions';
 
 function mapStateToProps(state: GlobalState) {
+    const config = state['plugins-jitsi'].config;
```
Comment (manojmalik20) [inline webapp/src/components/conference/index.ts]: Use a constant for the plugin id everywhere.

File: webapp/src/components/conference/index.ts [hunk 5]
```diff
@@ -8,10 +8,14 @@ import Conference from './conference';
...
+
     return {
         currentUserId: getCurrentUserId(state),
         post: state['plugins-jitsi'].openMeeting,
-        jwt: state['plugins-jitsi'].openMeetingJwt
+        jwt: state['plugins-jitsi'].openMeetingJwt,
+        showPrejoinPage: Boolean(config.show_prejoin_page),
```
Comment (manojmalik20) [inline webapp/src/components/conference/index.ts]:
```suggestion
        showPrejoinPage: config.show_prejoin_page,
```

File: webapp/src/types.ts [hunk 6]
```diff
@@ -18,3 +18,5 @@ export type GlobalState = ReduxGlobalState & {
         }
     }
 }
+
+export type plugin = 'plugins-jitsi'
```
Comment (manojmalik20) [inline webapp/src/types.ts:22]: You can use the constant here as well.

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/components/conference/conference.tsx [hunk 7]
```diff
@@ -139,7 +141,8 @@ export default class Conference extends React.PureComponent<Props, State> {
             const vh = this.getViewportHeight();
             const iframe = this.api.getIFrame();
             iframe.style.width = this.state.minimized ? MINIMIZED_WIDTH + 'px' : vw + 'px';
-            iframe.style.height = this.state.minimized ? MINIMIZED_HEIGHT + 'px' : vh + 'px';
+            const minimizedHeight = this.props.showPrejoinPage ? `calc(${WINDOW_HEIGHT}vh - ${MATTERMOST_HEADER_HEIGHT}px)` : MINIMIZED_HEIGHT + 'px';
```
Comment (manojmalik20) [inline webapp/src/components/conference/conference.tsx]:
```suggestion
            const minimizedHeight = this.props.showPrejoinPage ? `calc(${WINDOW_HEIGHT}vh - ${MATTERMOST_HEADER_HEIGHT}px)` : `${MINIMIZED_HEIGHT}px`;
```

File: webapp/src/components/conference/conference.tsx [hunk 8]
```diff
@@ -363,16 +367,17 @@ export default class Conference extends React.PureComponent<Props, State> {
     }
 }
 
-function getStyle(height: number, width: number, position: 'top' | 'bottom'): { [key: string]: React.CSSProperties } {
+function getStyle(height: number, width: number, position: 'top' | 'bottom', showPrejoinPage: boolean, isMinimized: boolean): { [key: string]: React.CSSProperties } {
     const backgroundZIndex = 1000;
     const jitsiZIndex = 1100;
     const buttonsZIndex = 1200;
     const loadingZIndex = 1200;
+    const minimizedPositionBottom = position === POSITION_BOTTOM ? ((height - BORDER_SIZE) - BUTTONS_PADDING_TOP) + 'px' : '';
```
Comment (manojmalik20) [inline webapp/src/components/conference/conference.tsx]:
```suggestion
    const minimizedPositionBottom = position === POSITION_BOTTOM ? `${height - BORDER_SIZE - BUTTONS_PADDING_TOP}px` : '';
```

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/types.ts (hunk 6 above)
Comment (raghavaggarwal2308) [reply webapp/src/types.ts:22]: discussed

**Pull request merged**